test/e2e/autoscale_test.go

```diff
@@ -202,6 +203,17 @@ func assertScaleUp(ctx *testContext) {
 	if err != nil {
 		ctx.t.Fatalf("Unable to observe the Deployment named %s scaling up. %s", ctx.deploymentName, err)
 	}
+
+	dep, err := ctx.clients.KubeClient.Kube.AppsV1().Deployments(test.ServingNamespace).Get(ctx.deploymentName, metav1.GetOptions{})
+	if err != nil {
+		ctx.t.Fatalf("Could not get deployment %v", ctx.deploymentName)
+	}
+
+	// verify scale-up has happened
+	if dep.Status.ReadyReplicas <= 0 {
```

Comment:
```suggestion
	if dep.Status.ReadyReplicas == 0 {
```

Reply:
Hmm ...   ReadyReplicas is not an unsigned int so it could go negative.   I guess it is always be >= 0.